bee-network/src/network/error.rs

```diff
@@ -3,19 +3,19 @@
 
 use crate::alias;
 
-use libp2p_core::{Multiaddr, PeerId};
+use libp2p::{swarm::DialError, Multiaddr, PeerId};
 
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
     #[error("Dialing address {} was denied.", .0)]
     DialingAddressDenied(Multiaddr),
 
-    #[error("Dialing address {} failed.", .0)]
-    DialingAddressFailed(Multiaddr),
+    #[error("Dialing address {} failed. Cause: {:?}", .0, .1)]
```

Comment:
I think you can do `#[error("Dialing address {0} failed. Cause: {1:?}")]`. Same for the others

Reply:
Yea, that's much nicer. Will do that!!